src/6model/reprs/P6opaque.c

```diff
@@ -893,17 +906,15 @@ static void deserialize_stable_size(MVMThreadContext *tc, MVMSTable *st, MVMSeri
     /* To calculate size, we need number of attributes and to know about
      * anything flattend in. */
     MVMint64  num_attributes = MVM_serialization_read_int(tc, reader);
-    MVMuint32 cur_offset = sizeof(MVMP6opaque);
+    MVMuint64 cur_offset = sizeof(MVMP6opaque);
     MVMint64  i;
     for (i = 0; i < num_attributes; i++) {
         if (MVM_serialization_read_int(tc, reader)) {
             MVMSTable *st = MVM_serialization_read_stable_ref(tc, reader);
             const MVMStorageSpec *ss = st->REPR->get_storage_spec(tc, st);
             if (ss->inlineable) {
                 /* TODO: Review if/when we get sub-byte things. */
-                if (cur_offset % ss->align) {
-                    cur_offset += ss->align - cur_offset % ss->align;
-                }
+                align_to(&cur_offset, (MVMuint32)ss->align);
```

Comment:
Here I have simply translated the `if (...) { ... }` idiom into the `align_to()`, but
shouldn't the alignment happen after `cur_offset +=`?

Reply:
It seems that this one it OK, there should however be an `align_to()` added before the write to `st->size`.